### PR TITLE
Use phusion:baseimage for glbc image

### PR DESCRIPTION
**Pull request by paralin**
Minimizes the size of the image without really sacrificing anything.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/229)

### CODE REVIEW

**Comment by bprashanth** [comment]
Can you please test this against a gce cluster with something like:
```
$kube-repo/hack/ginkgo-e2e.sh --ginkgo.focus=GCE.*L7.*
```

File: Ingress/controllers/gce/Dockerfile [hunk 1]
```diff
@@ -12,23 +12,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# TODO: use radial/busyboxplus:curl or alping instead
-FROM ubuntu:14.04
+FROM phusion:baseimage:0.9.17
```
Comment (bprashanth) [inline Ingress/controllers/gce/Dockerfile]: how about just going tiny image busybox?

File: Ingress/controllers/gce/Dockerfile [hunk 2]
```diff
@@ -12,23 +12,15 @@
...
 MAINTAINER Prashanth B <beeps@google.com>
+MAINTAINER Christian Stewart <christian@paral.in>
+CMD ["/sbin/my_init"]
```
Comment (bprashanth) [inline Ingress/controllers/gce/Dockerfile]: what's my_init?

File: Ingress/controllers/gce/Dockerfile [hunk 3]
```diff
@@ -12,23 +12,15 @@
...
-# an ubuntu container for dev purposes trumps the tiny amounts
-# of disk and bandwidth we'd save in doing so.
 RUN \
   apt-get update && \
-  apt-get install -y ca-certificates && \
-  apt-get install -y curl && \
-  rm -rf /var/lib/apt/lists/*
-
-ADD glbc glbc
-ENTRYPOINT ["/glbc"]
```
Comment (bprashanth) [inline Ingress/controllers/gce/Dockerfile]: why remove this? I mean you can, but you'll need to modify args and cmd in the yaml

File: Ingress/controllers/gce/Dockerfile [hunk 4]
```diff
@@ -12,23 +12,15 @@
...
+  apt-get install -y ca-certificates curl && \
+  apt-get clean && \
+  rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+RUN mkdir /etc/service/glbc
+ADD glbc /etc/service/glbc/run
```
Comment (bprashanth) [inline Ingress/controllers/gce/Dockerfile]: Did you mean to include the init script?

File: Ingress/controllers/gce/Dockerfile (hunk 1 above)
Comment (bprashanth) [inline Ingress/controllers/gce/Dockerfile]: Just a suggestion

File: Ingress/controllers/gce/Dockerfile (hunk 2 above)
Comment (paralin) [inline Ingress/controllers/gce/Dockerfile]: my_init is the phusion baseimage startup script

File: Ingress/controllers/gce/Dockerfile (hunk 4 above)
Comment (paralin) [inline Ingress/controllers/gce/Dockerfile]: Yeah init script should be there and executable in some other dir. Will fix.

File: Ingress/controllers/gce/Dockerfile (hunk 2 above)
Comment (bprashanth) [inline Ingress/controllers/gce/Dockerfile]: Please document this then. Just a comment will do.

**Comment by test-foxish** [comment]
recomputing cla status...

**Comment by test-foxish** [comment]
recomputing cla status...

**Comment by paralin** [comment]
I signed it!

**Comment by paralin** [comment]
Going to close this, though.

**Pull request closed without merging**